homeassistant/components/romy/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow for ROMY integration."""
+from __future__ import annotations
+
+import romy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN, LOGGER
+
+
+def _schema_with_defaults(
+    host: str = "", name: str = "", requires_password: bool = False
+) -> vol.Schema:
+    schema = {
+        vol.Required(CONF_HOST, default=host): cv.string,
+        vol.Optional(CONF_NAME, default=name): cv.string,
+    }
+
+    if requires_password:
+        schema.update(
+            {vol.Required(CONF_PASSWORD, default=""): vol.All(str, vol.Length(8))}
+        )
+
+    return vol.Schema(schema)
+
+
+class RomyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle config flow for ROMY."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Handle a config flow for ROMY."""
+        self.discovery_schema = None
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        errors: dict[str, str] = {}
+        data = self.discovery_schema or _schema_with_defaults()
+
+        if user_input is not None:
+            # Save the user input and finish the setup
+            host = user_input["host"]
+            name = user_input["name"]
+            password = ""
+            if "password" in user_input:
+                password = user_input["password"]
+
+            new_romy = await romy.create_romy(host, password)
+
+            # get robots name in case none was provided
+            if name == "":
+                name = new_romy.name
+                user_input["name"] = new_romy.name
+
+            if not new_romy.is_initialized:
+                errors[CONF_HOST] = "cannot_connect"
+
+            if not new_romy.is_unlocked:
+                data = _schema_with_defaults(
+                    host=host, name=name, requires_password=True
+                )
+                errors[CONF_PASSWORD] = "invalid_auth"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input["name"], data=user_input
```

Comment:
```suggestion
                    title=user_input[CONF_NAME], data=user_input
```

Please use constants whenever possible

Reply:
using constants now